src/Text/Parsing/Parser/Stream.purs

```diff
@@ -39,74 +40,74 @@ instance charHasUpdatePosition :: HasUpdatePosition Char where
 -- |
 -- | Instances must satisfy the following laws:
 -- | - `stripPrefix (Prefix a) a >>= uncons = Nothing`
-class StreamLike f c | f -> c where
-  uncons :: f -> Maybe { head :: c, tail :: f, updatePos :: Position -> Position }
-  stripPrefix :: Prefix f -> f -> Maybe {  rest :: f, updatePos :: Position -> Position }
+class StreamLike s m t | s -> t where
```

Comment:
not sure, in parsec this is how it looks `class (Monad m) => Stream s m t | s -> t`